gax/src/main/java/com/google/api/gax/retrying/RetryingFuture.java

```diff
@@ -34,27 +34,108 @@
 import java.util.concurrent.Callable;
 
 /**
- * Represents a retrying future. This is a facade hiding all the complications of an asynchronous
- * execution of a retriable task.
+ * Represents a retrying future. This is a facade hiding all the complications of an
+ * asynchronous/synchronous execution of a retriable task.
  *
  * <p>This interface is for advanced/internal use only.
  *
  * @param <ResponseT> response type
  */
 @BetaApi
 public interface RetryingFuture<ResponseT> extends ApiFuture<ResponseT> {
-
   /**
-   * Sets the attempt future. This future represents a concrete retry attempt, potentially scheduled
-   * for execution in a some form of {@link java.util.concurrent.ScheduledExecutorService}.
+   * Sets the attempt in a form of a future (asynchronous attempt). This future represents a
+   * concrete retry attempt, potentially scheduled for execution in a some form of {@link
+   * java.util.concurrent.ScheduledExecutorService}.
    *
    * @param attemptFuture the attempt future
    */
   void setAttemptFuture(ApiFuture<ResponseT> attemptFuture);
 
-  /** Returns current (active) attempt settings. */
-  TimedAttemptSettings getAttemptSettings();
+  /**
+   * Sets the attempt by providing its completed result (synchronous attempt).
+   *
+   * @param throwable throwable thrown by the attempt, {@code null} if the attempt returned response
+   *     instead
+   * @param response response returned by the attempt, {@code null} can be a valid response value
+   */
+  void setAttempt(Throwable throwable, ResponseT response);
```

Comment:
Please no. AttemptResult is a different thing, basically setAttempt calls setAttemptResult inside it (via same thread callback, or directly on BasicRetryingFuture.java#107